decoder/stats.go

```diff
@@ -246,21 +247,35 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areaNames []geo.AreaName) {
 
 			if old == nil || old.PokemonId != new.PokemonId { // pokemon is new or type has changed
 				countStats.count[new.PokemonId]++
+				external.PokemonCountNew.WithLabelValues(area.Parent).Inc()
+				if new.ExpireTimestampVerified {
+					external.UpdateVerifiedTtl(area, new.SeenType, new.ExpireTimestamp)
+				}
 			}
 			if new.Cp.Valid {
 				countStats.ivCount[new.PokemonId]++
+				external.PokemonCountIv.WithLabelValues(area.Parent).Inc()
 				if new.Shiny.ValueOrZero() {
+					external.PokemonCountShiny.WithLabelValues(area.Parent).Inc()
 					countStats.shiny[new.PokemonId]++
 				}
 				if new.AtkIv.Valid && new.DefIv.Valid && new.StaIv.Valid {
 					atk := new.AtkIv.ValueOrZero()
 					def := new.DefIv.ValueOrZero()
 					sta := new.StaIv.ValueOrZero()
 					if atk == 15 && def == 15 && sta == 15 {
+						external.PokemonCountHundo.WithLabelValues(area.Parent).Inc()
 						countStats.hundos[new.PokemonId]++
+						if new.Shiny.ValueOrZero() {
```

Comment:
If you want to count shinies this is not the place
You would need to do this in the basic encounter updator and also log the denominator - since shinies are per worker you don't actually get all of them here, or even all encounters to be able to know the ratio.
I'd probably suggest that you just drop the shiny counters entirely - but if you do want them, then put them somewhere where it'll actually see all shinies